scipy/optimize/lbfgsb.py

```diff
@@ -336,6 +336,8 @@ def func_and_grad(x):
         _lbfgsb.setulb(m, x, low_bnd, upper_bnd, nbd, f, g, factr,
                        pgtol, wa, iwa, task, iprint, csave, lsave,
                        isave, dsave, maxls)
+        # f should really always be a scalar
+        f = np.asarray(f).reshape(())
```

Comment:
I think this should be after line 347.

Better would be to raise a deprecationwarning there if f is not a scalar, and then apply this transformation.

Reply:
Done. This fix removed the need for some other workarounds btw.